### PR TITLE
[Fix](query execution) Fix result sink fragment can't be cancelled in non-pipeline

### LINKED ISSUE #25523: [Bug] Query cannot be cancelled
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

*

### What You Expected?

*

### How to Reproduce?

in test_array_map_function, test for non-pipeline will hang.

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
close https://github.com/apache/doris/issues/25523

Not merge to 2.0 branch

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run P0

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/pipeline/task_scheduler.cpp [hunk 1]
```diff
@@ -228,8 +228,6 @@ void TaskScheduler::_do_work(size_t index) {
         }
         task->set_task_queue(_task_queue.get());
         auto* fragment_ctx = task->fragment_context();
-        signal::query_id_hi = fragment_ctx->get_query_id().hi;
-        signal::query_id_lo = fragment_ctx->get_query_id().lo;
```
Comment (yiguolei) [inline be/src/pipeline/task_scheduler.cpp:232]: why delete this？

Comment (zhiqiang-hhhh) [reply be/src/pipeline/task_scheduler.cpp:232]: unused variable

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/runtime/fragment_mgr.cpp [hunk 2]
```diff
@@ -961,6 +961,14 @@ void FragmentMgr::_set_scan_concurrency(const Param& params, QueryContext* query
 #endif
 }
 
+bool FragmentMgr::is_pipeline_fragment(const TUniqueId& id) {
+    std::unique_lock<std::mutex> state_lock(_lock);
+    if (_pipeline_map.contains(id)) {
+        return true;
```
[the remark this answers is not in the document]
Comment (zhiqiang-hhhh) [reply be/src/runtime/fragment_mgr.cpp]: fixed

File: be/src/runtime/fragment_mgr.cpp [hunk 3]
```diff
@@ -951,6 +960,14 @@ void FragmentMgr::_set_scan_concurrency(const Param& params, QueryContext* query
...
```
[the remark this answers is not in the document]
Comment (zhiqiang-hhhh) [reply be/src/runtime/fragment_mgr.cpp]: fixed

File: be/src/service/internal_service.cpp [hunk 4]
```diff
@@ -558,15 +558,19 @@ void PInternalServiceImpl::cancel_plan_fragment(google::protobuf::RpcController*
...
-            LOG(INFO) << "cancel fragment, fragment_instance_id=" << print_id(tid)
-                      << ", reason: " << PPlanFragmentCancelReason_Name(request->cancel_reason());
-            _exec_env->fragment_mgr()->cancel_instance(tid, request->cancel_reason());
-        } else {
-            LOG(INFO) << "cancel fragment, fragment_instance_id=" << print_id(tid);
-            _exec_env->fragment_mgr()->cancel_instance(tid,
-                                                       PPlanFragmentCancelReason::INTERNAL_ERROR);
-        }
+
+        const bool is_pipeline_fragment = _exec_env->fragment_mgr()->is_pipeline_fragment(tid);
```
Comment (yiguolei) [inline be/src/service/internal_service.cpp:562]: 把这个方法删了吧，加一个pipeline的前缀意义不大
可以在cancel insantance unlock 里面加pipeline的信息

File: be/src/runtime/fragment_mgr.cpp [hunk 5]
```diff
@@ -961,6 +961,11 @@ void FragmentMgr::_set_scan_concurrency(const Param& params, QueryContext* query
 #endif
 }
 
+bool FragmentMgr::is_pipeline_fragment(const TUniqueId& id) {
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:964]: 把这个方法删除了吧

File: be/src/runtime/fragment_mgr.cpp [hunk 6]
```diff
@@ -979,37 +984,13 @@ void FragmentMgr::cancel_query_unlocked(const TUniqueId& query_id,
...
-            // so we add this check to avoid too many WARNING log.
-            if (_pipeline_map.contains(it)) {
-                cancel_instance_unlocked(it, reason, state_lock, msg);
-            }
-        }
-    } else {
-        for (auto it : ctx->second->fragment_ids) {
-            cancel_fragment_unlocked(it, reason, state_lock, msg);
-        }
+    for (auto it : ctx->second->fragment_ids) {
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:987]: rename fragment_ids to fragment_instance_ids

File: be/src/service/internal_service.cpp [hunk 7]
```diff
@@ -558,15 +558,19 @@ void PInternalServiceImpl::cancel_plan_fragment(google::protobuf::RpcController*
...
+        const bool has_cancel_reason = request->has_cancel_reason();
+        LOG(INFO) << fmt::format("Cancel {}instance {}, reason: {}",
+                                 is_pipeline_fragment ? "pipeline " : "", print_id(tid),
+                                 has_cancel_reason
+                                         ? PPlanFragmentCancelReason_Name(request->cancel_reason())
+                                         : "INTERNAL_ERROR");
+
+        _exec_env->fragment_mgr()->cancel_instance(
```
Comment (yiguolei) [inline be/src/service/internal_service.cpp:570]: rename cancel_instance to  cancel_fragment_instance

File: be/src/runtime/fragment_mgr.cpp (hunk 5 above)
Comment (zhiqiang-hhhh) [reply be/src/runtime/fragment_mgr.cpp]: rm done

File: be/src/runtime/fragment_mgr.cpp (hunk 6 above)
Comment (zhiqiang-hhhh) [reply be/src/runtime/fragment_mgr.cpp]: done

File: be/src/service/internal_service.cpp (hunk 7 above)
Comment (zhiqiang-hhhh) [reply be/src/service/internal_service.cpp:568]: done

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**